apps/ledger-live-desktop/src/renderer/modals/SignMessage/Components/CopyButton.test.tsx

```diff
@@ -0,0 +1,101 @@
+import React from "react";
+import { render, screen, fireEvent, waitFor } from "@testing-library/react";
+import { DefaultTheme, ThemeProvider } from "styled-components";
+import CopyButton from "./CopyButton";
+
+jest.mock("i18next", () => ({
+  t: jest.fn(key => {
+    if (key === "common.copy") return "Copy";
+    if (key === "common.copied") return "Copied";
+    return key;
+  }),
+}));
+
+const mockTheme = {
+  colors: {
+    neutral: {
+      c30: "#F4F4F4",
+      c90: "#4D4D4D",
+    },
+    success: {
+      c50: "#5BA54C",
+    },
+  },
+} as DefaultTheme;
+
+jest.mock("@ledgerhq/react-ui", () => ({
+  __esModule: true,
+  Button: ({ children, onClick }: never) => (
+    <button onClick={onClick} data-testid="ledger-button">
+      {children}
+    </button>
+  ),
+  Icons: {
+    Check: () => <div data-testid="check-icon" />,
+    Copy: () => <div data-testid="copy-icon" />,
+  },
+}));
+
+beforeAll(() => {
+  Object.assign(navigator, {
+    clipboard: {
+      writeText: jest.fn().mockResolvedValue(undefined),
+    },
+  });
+});
+
+afterAll(() => {
+  jest.restoreAllMocks();
+});
+
+describe("CopyButton", () => {
+  const testText = "Text to copy";
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  const renderWithTheme = (component: React.ReactNode) => {
+    return render(<ThemeProvider theme={mockTheme}>{component}</ThemeProvider>, {
+      wrapper: ({ children }) => children,
+    });
+  };
+
+  it("renders correctly with copy icon and text", async () => {
+    const { unmount } = renderWithTheme(<CopyButton text={testText} />);
+
+    expect(screen.getByTestId("copy-icon")).toBeInTheDocument();
+    expect(screen.getByText("Copy")).toBeInTheDocument();
+
+    unmount();
+  });
+
+  it("copies text to clipboard when clicked", async () => {
+    const { unmount } = renderWithTheme(<CopyButton text={testText} />);
+
+    await waitFor(() => {
+      fireEvent.click(screen.getByTestId("ledger-button"));
+    });
```

Comment:
```suggestion
    const { user } = renderWithTheme(<CopyButton text={testText} />);

    const btn = screen.getByRole("button","Copy");
    expect(btn).toBeVisible();
    await user.click(btn);
```

you can check this doc : https://testing-library.com/docs/queries/about/#priority